mod/developers/actions/developers/ajax_demo.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+// example of altering response object via hook. This might be done in a plugin
+
+function developers_ajax_demo_alter($hook, $type, Elgg\Services\AjaxResponse $v, $p) {
+	// check data added by client hook
+	if (get_input('client_request_altered') == '1') {
+		// add some data to the response
+		$v->getData()->server_response_altered = 2;
+	}
+
+	register_error('Error from ajax demo response hook');
+}
+elgg_register_plugin_hook_handler('ajax_response', 'action:developers/ajax_demo', 'developers_ajax_demo_alter');
```

Comment:
Yep, filtering the action output after the action completes. But that's another story and another ticket.